core/src/main/java/org/apache/gravitino/authorization/AuthorizationUtils.java

```diff
@@ -364,4 +376,123 @@ private static void checkCatalogType(
+        case METALAKE:
+          {
+            NameIdentifier[] identifiers =
+                GravitinoEnv.getInstance().catalogDispatcher().listCatalogs(Namespace.of(metalake));
+            Arrays.stream(identifiers)
+                .collect(Collectors.toList())
+                .forEach(
+                    identifier -> {
+                      Catalog catalogObj =
+                          GravitinoEnv.getInstance().catalogDispatcher().loadCatalog(identifier);
+                      if (catalogObj.provider().equals("hive")) {
+                        Schema schema =
+                            GravitinoEnv.getInstance()
+                                .schemaDispatcher()
+                                .loadSchema(
+                                    NameIdentifier.of(
+                                        metalake,
+                                        catalogObj.name(),
+                                        "default" /*Hive default schema*/));
+                        if (schema.properties().containsKey(HiveConstants.LOCATION)) {
+                          String defaultSchemaLocation =
+                              schema.properties().get(HiveConstants.LOCATION);
+                          Preconditions.checkArgument(
+                              defaultSchemaLocation != null,
+                              String.format("Catalog %s location is not found", ident));
+                          locations.add(defaultSchemaLocation);
+                        }
+                      }
+                    });
+          }
+          break;
+        case CATALOG:
+          {
+            Catalog catalogObj = GravitinoEnv.getInstance().catalogDispatcher().loadCatalog(ident);
+            if (catalogObj.provider().equals("hive")) {
+              Schema schema =
+                  GravitinoEnv.getInstance()
+                      .schemaDispatcher()
+                      .loadSchema(
+                          NameIdentifier.of(
+                              metalake, catalogObj.name(), "default" /*Hive default schema*/));
+              if (schema.properties().containsKey(HiveConstants.LOCATION)) {
+                String defaultSchemaLocation = schema.properties().get(HiveConstants.LOCATION);
+                Preconditions.checkArgument(
+                    defaultSchemaLocation != null,
+                    String.format("Catalog %s location is not found", ident));
+                locations.add(defaultSchemaLocation);
+              }
+            }
+          }
+          break;
+        case SCHEMA:
+          {
+            Schema schema = GravitinoEnv.getInstance().schemaDispatcher().loadSchema(ident);
+            if (schema.properties().containsKey(HiveConstants.LOCATION)) {
+              String schemaLocation = schema.properties().get(HiveConstants.LOCATION);
+              Preconditions.checkArgument(
+                  schemaLocation != null, String.format("Schema %s location is not found", ident));
```

Comment:
schemaLocation could be null. If you don't set location, the location will use the default location.